### PR TITLE
:wrench: s3 decouple from spring oauth setup

### PR DESCRIPTION
**Pull request by simonhir**
**Description**

Decouple s3-rest-client and s3-rest-service from Spring OAuth2 setup by removing default setup and adding it as example to the README.
Fixes coupling to specific scopes and naming.

### CODE REVIEW

File: refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml [hunk 1]
```diff
@@ -6,6 +6,25 @@ spring:
...
+            user-name-attribute: user_name
+        registration:
+          s3:
+            provider: keycloak
+            authorization-grant-type: client_credentials
+            client-id: local
+            client-secret: client_secret
+            # profile required for username used in s3-rest-service RequestResponseLoggingFilter
+            # openid required for user info endpoint used in s3-rest-service JwtUserInfoAuthenticationConverter
+            scope: profile, openid
```
Comment (boal) [inline refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml:27]: The values within the scope property are  LHM specific. The comment above should mention, that the scope values are an example

File: refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml [hunk 2]
```diff
@@ -6,6 +6,25 @@ spring:
...
+  security:
+    oauth2:
+      client:
+        provider:
+          keycloak:
+            issuer-uri: http://keycloak:8100/auth/realms/local_realm
+            user-info-uri: ${spring.security.oauth2.client.provider.keycloak.issuer-uri}/protocol/openid-connect/userinfo
+            jwk-set-uri: ${spring.security.oauth2.client.provider.keycloak.issuer-uri}/protocol/openid-connect/certs
+            # used for RequestResponseLoggingFilter in s3-rest-service
+            user-name-attribute: user_name
```
Comment (boal) [inline refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml:18]: The RequestResponseLoggingFilter and the corresponding  property `user-name-attribute` are LHM specific. Both should be removed from the integration lib.

File: refarch-integrations/refarch-s3-integration/README.md [hunk 3]
```diff
@@ -65,6 +80,29 @@ All properties of [s3-integration-java-client-starter](#s3-integration-rest-clie
...
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          keycloak:
+            issuer-uri: https://sso.example.com/auth/realms/refarch
+            user-info-uri: ${spring.security.oauth2.client.provider.keycloak.issuer-uri}/protocol/openid-connect/userinfo
+            jwk-set-uri: ${spring.security.oauth2.client.provider.keycloak.issuer-uri}/protocol/openid-connect/certs
+            # used for RequestResponseLoggingFilter in s3-rest-service
```
Comment (boal) [inline refarch-integrations/refarch-s3-integration/README.md:97]: The RequestResponseLoggingFilter and the corresponding property user-name-attribute are LHM specific. Both should be removed from the integration lib.

File: refarch-integrations/refarch-s3-integration/README.md [hunk 4]
```diff
@@ -65,6 +80,29 @@ All properties of [s3-integration-java-client-starter](#s3-integration-rest-clie
...
+            user-name-attribute: user_name
+        registration:
+          s3:
+            provider: keycloak
+            authorization-grant-type: client_credentials
+            client-id: refarch_client
+            client-secret: client_secret_123
+            # profile required for username used in s3-rest-service RequestResponseLoggingFilter
+            # openid required for user info endpoint used in s3-rest-service JwtUserInfoAuthenticationConverter
+            scope: profile, openid
```
Comment (boal) [inline refarch-integrations/refarch-s3-integration/README.md:107]: The values within the scope property are LHM specific. The comment above should mention, that the scope values are an example

**Review by boal — CHANGES_REQUESTED** [review]

File: refarch-integrations/refarch-s3-integration/README.md [hunk 5]
```diff
@@ -65,6 +80,29 @@ All properties of [s3-integration-java-client-starter](#s3-integration-rest-clie
...
+For authentication against the s3-service a OAuth2 registration with the name `s3` needs to be provided.
+See following example or the [according Spring documentation](https://docs.spring.io/spring-security/reference/servlet/oauth2/index.html#oauth2-client).
+
+```yml
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          keycloak:
```
Comment (boal) [inline refarch-integrations/refarch-s3-integration/README.md:93]: the provider name `keycloak` should not be product specific. 

An alternative would be `sso` or an other appropriate name.

File: refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml [hunk 6]
```diff
@@ -6,6 +6,25 @@ spring:
     port: 1025
     username: test@muenchen.de
     password: secret
+  security:
+    oauth2:
+      client:
+        provider:
+          keycloak:
```
Comment (boal) [inline refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml:13]: the provider name keycloak should not be product specific.

An alternative would be sso or an other appropriate name.

**Review by boal — CHANGES_REQUESTED** [review]

File: refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml (hunk 1 above)
Comment (simonhir) [reply refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml]: Removed as not needed by test.

File: refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml (hunk 2 above)
Comment (simonhir) [reply refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml]: Removed as not needed by test.

File: refarch-integrations/refarch-s3-integration/README.md (hunk 4 above)
Comment (simonhir) [reply refarch-integrations/refarch-s3-integration/README.md:109]: I added, that these are only required when explicitly enabling the mentioned featues.

File: refarch-integrations/refarch-s3-integration/README.md (hunk 3 above)
Comment (simonhir) [reply refarch-integrations/refarch-s3-integration/README.md:97]: As these come from the refarch-templates and this is a general concern i won't fix it in this MR.
See https://github.com/it-at-m/refarch-templates/issues/354

File: refarch-integrations/refarch-s3-integration/README.md (hunk 5 above)
Comment (simonhir) [reply refarch-integrations/refarch-s3-integration/README.md]: Done

File: refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml (hunk 6 above)
Comment (simonhir) [reply refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml]: Done

**Review by boal — APPROVED** [review]

File: refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml [hunk 7]
```diff
@@ -6,6 +6,20 @@ spring:
     port: 1025
     username: test@muenchen.de
     password: secret
+  security:
+    oauth2:
+      client:
+        provider:
+          sso:
+            issuer-uri: http://keycloak:8100/auth/realms/local_realm
+            user-info-uri: ${spring.security.oauth2.client.provider.keycloak.issuer-uri}/protocol/openid-connect/userinfo
```
Comment (devtobi) [inline refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml:15]: Where is `${spring.security.oauth2.client.provider.keycloak.issuer-uri}` coming from?

File: refarch-integrations/refarch-s3-integration/README.md [hunk 8]
```diff
@@ -65,6 +80,31 @@ All properties of [s3-integration-java-client-starter](#s3-integration-rest-clie
...
+
+```yml
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          sso:
+            issuer-uri: https://sso.example.com/auth/realms/refarch
+            user-info-uri: ${spring.security.oauth2.client.provider.keycloak.issuer-uri}/protocol/openid-connect/userinfo
```
Comment (devtobi) [inline refarch-integrations/refarch-s3-integration/README.md:95]: Where is ${spring.security.oauth2.client.provider.keycloak.issuer-uri} coming from?

File: refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml (hunk 7 above)
Comment (simonhir) [reply refarch-integrations/refarch-email-integration/refarch-email-integration-rest-example/src/main/resources/application-local.yml]: Fixed, i forgot to rename that when renaming the parent key

File: refarch-integrations/refarch-s3-integration/README.md (hunk 8 above)
Comment (simonhir) [reply refarch-integrations/refarch-s3-integration/README.md]: Fixed, i forgot to rename that when renaming the parent key

File: refarch-integrations/refarch-s3-integration/README.md [hunk 9]
```diff
@@ -47,7 +47,22 @@ Whether a property is an alias can be checked in the corresponding `application.
...
+See below example or the [according Spring documentation](https://docs.spring.io/spring-security/reference/servlet/oauth2/index.html#oauth2-resource-server).
+
+```yml
+spring:
+  security:
+    oauth2:
+      resourceserver:
+        jwt:
+          issuer-uri: https://sso.example.com/auth/realms/refarch
+security:
```
[the remark this answers is not in the document]
Comment (simonhir) [reply refarch-integrations/refarch-s3-integration/README.md:61]: Indentation is correct as it comes from the refarch-backend. See https://github.com/it-at-m/refarch-templates/blob/main/refarch-backend/src/main/java/de/muenchen/refarch/configuration/SecurityConfiguration.java#L30
Should probably also start with `refarch` but think that's out of scope of this ticket.

[the remark this answers is not in the document]
Comment (devtobi) [reply refarch-integrations/refarch-s3-integration/README.md:61]: Ok, will be futher handled in https://github.com/it-at-m/refarch-templates/issues/355 and then migrated to this repository.

**Review by devtobi — APPROVED** [review]
LGTM

**Pull request merged**